src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -215,6 +241,15 @@ public CartDTO getCartByEmail(String email) throws EntityNotFoundException {
 
         Cart cart = cartRepository.findByUserId(user.getId()).orElseThrow(() ->
                 new EntityNotFoundException("Cart not found in the Database!"));
-            return cartEntityToDto(cart);
+        cartRepository.save(cart);
+
+        List<String> userCartEntries = cart.getEntries().stream().map(CartEntry::getProductName).collect(Collectors.toList());
+
+        String productParam = String.join(",", userCartEntries);
+
+        List<ProductDTO> productsInCart =  productServiceClient.callGetAllProductsFromUser(productParam, false);
```

Comment:
you could extract this logic into a separate method(since you use it for 4 times). something to receive cartEntries and to return a list of product dtos.

Reply:
done! good point here aswell! 👍 👍